etc/systemd/system/zfs-import-scan.service.in

```diff
@@ -13,6 +13,21 @@ ConditionPathIsDirectory=/sys/module/zfs
 [Service]
 Type=oneshot
 RemainAfterExit=yes
+EnvironmentFile=@initconfdir@/zfs
+
+# THIS SLEEP IS NECESSARY FOR IMPORT TO WORK
+# /etc/default/zfs adds -d /dev/disk/by-id flag but this directory gets
+# set up by systemd-udevd later during the boot process
+# You can't set a requirement "After udevd has finished the initial scan"
+# because the controllers can be slow, hotplugging, drive decryption,
+# lvm and so on can trigger new events
+
+# An alternative would be something like this
+# https://github.com/openzfs/zfs/issues/12966#issuecomment-1039437345
+# but I think it's solving a different issue?
+
+# 10s was claimed too little by systemd devs.
```

Comment:
Delaying pools import by 30 seconds for all users is rather unacceptable. Would you mind linking relevant discussions from systemd representatives?

Reply:
Ain't nothing I love more than delaying boot by *a full minute*. Hell, even on my laptop with one pool – so thirty seconds – this would make the boot take.. 30 times longer. We already do something similar in i-t, but there we import/sleep in a loop, but that's because we know we need *some*thing. I thought we arrived at "if you have a very slow device then add a sleep in an override or depend on it" a long time ago, this is ridiculous.

Reply:
Sorry, this part got submitted by accident. It shouldn't be a part of this PR at all.

Reply:
fixed